### PR TITLE
Allow one more origin. Log the reason of occ controller failure

### LINKED ISSUE #25973: Make cli version of PHP used by Updater configurable
### Steps to reproduce
1. Install (or update to) 9.0.4 OR 9.1.0 on a server running Ubuntu 12.04 LTS OR 14.04 LTS
2. It must also be running under Plesk 12.5 with alternative PHP version 7.0
3. The GUI is powered by Nginx + PHP-FPM, using the Plesk-PHP 7.0.9
4. Call Updater from the Web Admin page

### Expected behaviour
The Updater should work, and use the PHP 7.0.9, which is available on the host

### Actual behaviour
The Updater says: [UnexpectedValueException]
Could not parse a response for config:list. Please check if the current shell user can run occ command. Raw output: Not allowed

So it looks to me as if the Updater is using the standard OS-PHP, which no doubt cannot understand the PHP of OC.

My wish/suggestion is: To be able to add a config statement re the version of PHP to use for executing external PHP commands. No doubt you know that here in Germany many Hosters like HostEurope, Strato, Hetzner etc. offer Plesk, so it would be great to have this option. Otherwise, I have to tell my clients that they need a new server - something they don't want to hear )-:
UPDATE: Turns out, on later trials with Ubuntu 14.04, that PHP 5.3 was NOT the problem.

### Server configuration
Ubuntu 12.04 OR 14.04 LTS

**Web server:**
Nginx (from Plesk - 1.11.1)

**Database:**
MySQL

**PHP version:**
OS = 5.3.10 / 5.5.9, Plesk = 7.0.9 (at /opt/plesk/php/7.0/bin/php)

**ownCloud version:**
9.0.4 OR 9.1.0

**Updated from an older ownCloud or fresh install:**
Updated

**Where did you install ownCloud from:**
From Source-Zip, via CLI

**Signing status (ownCloud 9.0 and above):**
No errors have been found.

**List of activated apps:**
Enabled:
  - activity: 2.2.1
  - calendar: 1.3.3
  - contacts: 1.3.1.0
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - files: 1.4.4
  - files_external: 0.5.2
  - files_trashbin: 0.8.0
  - notifications: 0.2.3
  - updatenotification: 0.1.0

Disabled:
  - comments
  - encryption
  - external
  - federation
  - files_pdfviewer
  - files_sharing
  - files_texteditor
  - files_versions
  - files_videoplayer
  - firstrunwizard
  - gallery
  - provisioning_api
  - systemtags
  - templateeditor
  - user_external
  - user_ldap

**The content of config/config.php:**
{
    "system": {
        "instanceid": "oc452bf87cea",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.biblische-reisen.de"
        ],
        "datadirectory": "\/var\/www\/vhosts\/biblische-reisen.de\/oc-data",
        "dbtype": "mysql",
        "version": "9.0.4.1",
        "dbname": "owncloudbde",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 4,
        "forcessl": true,
        "theme": "",
        "maintenance": false,
        "mail_from_address": "owncloud",
        "mail_domain": "biblische-reisen.de",
        "mail_smtpmode": "php",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "default_language": "en",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "log_rotate_size": 10485760,
        "check_for_working_wellknown_setup": false,
        "check_for_working_htaccess": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0,
            "dbindex": 1
        },
        "trashbin_retention_obligation": "auto",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

**Are you using external storage**
No

**Are you using encryption:**
No

**Are you using an external user-backend, if yes which one:**
No

### Client configuration
**Browser:**
Firefox

**Operating system:**
Windows 10

#### Web server error log
```
Insert your webserver log here
Log is empty.

#### ownCloud log (data/owncloud.log)
Log is empty.

#### Browser log
17:55:21.680 Content Security Policy: Die Direktive 'frame-src'  sollte nicht mehr verwendet werden. Bitte verwenden Sie stattdessen die Direktive 'child-src'.(unbekannt)
17:55:23.842 Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src https://owncloud.biblische-reisen.de 'unsafe-eval'").admin

### PR DESCRIPTION
**Pull request by VicDeo**
## Description

- Allows requests to occ controller from $_SERVER['SERVER_ADDR']
- Adds more logging for non-authorized requests
## Related Issue

Fixes https://github.com/owncloud/core/issues/25973

## Motivation and Context

Allows to determine what precondition has been failed while opening web updater
## How Has This Been Tested?

By sending POST requests to `/index.php/occ/status` 
Any failed precondition should be written to owncloud.log
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/Controller/OccController.php [hunk 1]
```diff
@@ -108,6 +108,13 @@ public function execute($command, $token, $params = []) {
 			];
 
 		} catch (\UnexpectedValueException $e){
+			\OC::$server->getLogger()->warning(
```
Comment (DeepDiver1975) [inline core/Controller/OccController.php]: Inject logger

**Comment by VicDeo** [comment]
@DeepDiver1975 injected

File: core/Controller/OccController.php [hunk 2]
```diff
@@ -123,8 +135,13 @@ public function execute($command, $token, $params = []) {
 	 * @param $token
 	 */
 	protected function validateRequest($command, $token){
-		if (!in_array($this->request->getRemoteAddress(), ['::1', '127.0.0.1', 'localhost'])) {
-			throw new \UnexpectedValueException('Web executor is not allowed to run from a different host');
+		$allowedHosts = ['::1', '127.0.0.1', 'localhost'];
+		if (isset($this->request->server['SERVER_ADDR'])){
```
Comment (DeepDiver1975) [inline core/Controller/OccController.php]: Whats the deeper sense in this code block? This will result that any request is allowed - right?

Comment (DeepDiver1975) [inline core/Controller/OccController.php]: aha - getRemoteAddress va _SERVER['SERVER_ADDR'] - tooo early - need more :coffee:

Comment (VicDeo) [inline core/Controller/OccController.php]: @DeepDiver1975 SERVER_ADDR is not REMOTE_ADDR 
http://php.net/manual/en/reserved.variables.server.php
'SERVER_ADDR' The IP address of the server under which the current script is executing.
`getRemoteAddress` is used too but three lines below.

Comment (VicDeo) [inline core/Controller/OccController.php]: see also https://snippets.khromov.se/safe-php-_server-variables/

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
Backport?

**Comment by DeepDiver1975** [comment]
> Backport?

@PVince81 @VicDeo THX

**Comment by PVince81** [comment]
make sure it works in all envs, I have the feeling it won't where sometimes the internal host name is not the same as the external one. See test from @davitol where it did NOT work here: https://github.com/owncloud/updater/issues/378#issuecomment-251352361

**Comment by VicDeo** [comment]
Stable9: https://github.com/owncloud/core/issues/26276
Stable9.1: https://github.com/owncloud/core/issues/26277

**Pull request merged**